src/atoms/Text/Text.web.js

```diff
@@ -1,3 +1,140 @@
-const Text = () => null;
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import baseTheme from '../../tokens/theme';
+import { getColorKeys, getColor, getLineHeight } from '../../_helpers/theme';
+
+/*
+ * Sets height for the div
+ * @param {Object} theme
+ * @param {String} _lineHeight
+ * @param {number} maxLines
+ */
+const calculateHeight = ({ theme, _lineHeight, maxLines }) => {
+  const lineHeight = `${parseFloat(theme.fonts.lineHeight[_lineHeight])}`;
+  const containerHeight = `${lineHeight * maxLines}px`;
+
+  return containerHeight;
+};
+
+const styles = {
+  fontFamily({ theme, _weight }) {
+    return theme.fonts.family.lato[_weight];
+  },
+  fontSize({ theme, size }) {
+    return theme.fonts.size[size];
+  },
+  color({ theme, color }) {
+    return getColor(theme, color);
+  },
+  align({ align }) {
+    switch (align) {
+      case 'left':
+      default:
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+    }
+  },
+  letterSpacing({ theme, _letterSpacing }) {
+    return `${theme.fonts.letterSpacing[_letterSpacing]}`;
+  },
+  lineHeight({ theme, size, _lineHeight }) {
+    if (_lineHeight) {
+      return theme.fonts.lineHeight[_lineHeight];
+    } else {
+      return getLineHeight(theme, size);
+    }
+  },
+  maxHeight({ theme, _lineHeight, maxLines }) {
+    if (maxLines) {
+      const maxHeight = calculateHeight({ theme, _lineHeight, maxLines });
+      return maxHeight;
+    } else {
+      return 'initial';
+    }
+  },
+  whiteSpace({ maxLines }) {
+    if (maxLines) {
+      return 'initial';
+    } else {
+      return 'nowrap';
+    }
+  },
+  overflow({ maxLines }) {
+    if (maxLines) {
+      return 'hidden';
+    } else {
+      return 'initial';
+    }
+  },
+  textOverflow({ maxLines }) {
+    if (maxLines) {
```

Comment:
```suggestion
    if (isDefined(maxLines)) {
```

Reply:
is it a helper function to check if it is defined or not ?
Also if ( maxLines ) should work for the undefined values.

Reply:
```maxLines``` is a number so ```if(0)``` ```if(1)``` will act as boolean checks

Reply:
if (0) isn't false ? other numbers true ?

Reply:
```if(0)``` is false. The purpose of this check is not to treat the value of maxLines as a boolean check but to see if the prop is passed by the consumer and take action accordingly

Reply:
@deorukhkarchaitanya 
if (typeof maxLines !== 'undefined')
You are suggesting something like this ?

Reply:
No, I'm suggesting:

```
    if (isDefined(maxLines)) {
```

Reply:
sorry i didn't get you,  what is isDefined  ?

Reply:
done